java/arcs/core/data/testdata/IngestProcessRetrieve.arcs

```diff
@@ -0,0 +1,79 @@
+// Big example for recipe2plan generation.
+
+meta
+  namespace: arcs.core.data.testdata.gen
+
+schema Event
+    type: Text
+    timestamp: Number
+    value: Text
+
+particle Ingestion
+  events: writes [Event {type, timestamp, value}]
+
+// Example: Long running arc ingesting events into a collection.
+@trigger
+  launch startup
+  arcId eventIngestion
+recipe EventsIngestion
+  // Example: Handle created with id, capabilities and TTL.
+  events: create persistent 'raw-events' @ttl(1d)
+  Ingestion
+    events: events
+
+schema AnalyzedEvent
+  sentiment: Number
+  cohortIndex: Number
+
+particle AnalyzeSentiment
+  events: reads [Event {type, value}]
+  processed: writes [AnalyzedEvent {sentiment, cohortIndex}]
+
+particle SummarizeSentiment
+  pieces: reads [AnalyzedEvent {sentiment, cohortIndex}]
+  result: writes OverallSentiment {
+    overallSentiment: Number,
+    sentimentCohortBreakdown: [Number],
+    summary: Text
+  }
+
+// Example: Long running arc mapping data from another long running arc
+//          and providing data derived from the mapped data.
+@trigger
+  launch startup
+  arcId sentimentAnalysis
+recipe SentimentAnalyzer
+  // Example: Mapping by id and type; mapping a collection.
+  events: map 'raw-events'
+  // Example: Handle created without capabilities in a long running arc.
+  processed: create
+  summary: create 'events-summary' @ttl(15d)
+  AnalyzeSentiment
+    events: reads events
+    processed: writes processed
+  SummarizeSentiment
+    pieces: reads processed
+    result: writes summary
+
+particle RequestProvider
+  request: writes SentimentRequest {cohortId: Number}
+
+particle ResponseCalculator
+  request: reads SentimentRequest {cohortId: Number}
+  sentimentPerCohort: reads OverallSentiment {sentimentCohortBreakdown: [Number]}
+  response: writes SentimentResponse {sentimentValue: Number}
+
+// Example: Ephemeral arc mapping data from long running arc.
+recipe RequestHandler
+  // Example: Mapping through type alone; mapping a singleton.
+  sentimentSummary: map 'events-summary'
+  // Example: Handle created without capabilities in an ephemeral arc.
+  request: create
+  // Example: Handle created with capabilities in an ephemeral arc.
+  response: create tied-to-arc
```

Comment:
As of #5117, this should work :)

Reply:
It will likely put the right capability, but there is still some resolution issue we're hitting.